### PR TITLE
MM 20404 Migrate 'components/announcement_bar/version_bar' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by tianlangwu**
#### Summary
Migrate 'components/announcement_bar/version_bar' module and associated tests to TypeScript.
I am a beginner, not sure I did it correctly, please let me know where I did it wrong. 
I can't find where is "version_bar" in my browser to check is my code actually working.
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/16167
JIRA https://mattermost.atlassian.net/browse/MM-20404

### CODE REVIEW

File: components/announcement_bar/version_bar/version_bar.tsx [hunk 1]
```diff
@@ -11,20 +10,24 @@ import {AnnouncementBarTypes} from 'utils/constants';
 
 import AnnouncementBar from '../default_announcement_bar';
 
-export default class VersionBar extends React.PureComponent {
-    static propTypes = {
-        serverVersion: PropTypes.string,
-    };
+interface IMyComponentProps {
```
Comment (crspeller) [inline components/announcement_bar/version_bar/version_bar.tsx:13]: Usually we name these just `Props` and `State` if it's not exported, there is no need to be specific.

File: components/announcement_bar/version_bar/version_bar.tsx [hunk 2]
```diff
@@ -11,20 +10,24 @@ import {AnnouncementBarTypes} from 'utils/constants';
...
+    serverVersion?: string,
```
Comment (crspeller) [inline components/announcement_bar/version_bar/version_bar.tsx:14]: Are these really optional? I think we might be able to get away without the `?`

File: components/announcement_bar/version_bar/index.ts [hunk 3]
```diff
@@ -3,9 +3,9 @@
 
 import {connect} from 'react-redux';
 
-import VersionBar from './version_bar.jsx';
+import VersionBar from './version_bar';
 
-function mapStateToProps(state) {
+function mapStateToProps(state:any) {
```
Comment (crspeller) [inline components/announcement_bar/version_bar/index.ts:8]: Instead of `any` we can do `GlobalState` form 'types/store`

**Review by crspeller — CHANGES_REQUESTED** [review]

**Comment by tianlangwu** [comment]
Hi, thanks for the advice! I have fixed the three problems you mentioned. Could you tell me where is "version_bar" in the web app preview?

**Comment by crspeller** [comment]
@tianlangwu The version bar appears when a new version of the server is detected. I think you can make it appear by updating your sever to a new version without refreshing the browser.

**Review by crspeller — APPROVED** [review]

**Review by agarciamontoro — APPROVED** [review]
Thank you for your contribution, @tianlangwu! This looks good to me :tada:

**Review by jgilliam17 — APPROVED** [review]
Thank you @tianlangwu 
Tested, look good to merge.
- Verified new version available announcement banner - no change in functionality.
@agarciamontoro can you please help merge? Thanks 🙂

**Pull request merged**